src/notation/internal/notationactioncontroller.cpp

```diff
@@ -932,14 +932,22 @@ void NotationActionController::move(MoveDirection direction, bool quickly)
             interaction->moveLyrics(direction);
         } else if (selectedElement && (selectedElement->isTextBase() || selectedElement->isArticulation())) {
             interaction->nudge(direction, quickly);
-        } else if (interaction->noteInput()->isNoteInputMode()
-                   && interaction->noteInput()->state().staffGroup == mu::engraving::StaffGroup::TAB) {
-            interaction->moveSelection(direction, MoveSelectionType::String);
-            return;
         } else if (interaction->selection()->isNone()) {
             interaction->selectFirstElement(false);
         } else {
+            bool isTab = interaction->noteInput()->isNoteInputMode()
+                         && interaction->noteInput()->state().staffGroup == mu::engraving::StaffGroup::TAB;
+            int ostring = isTab && selectedElement->isNote() ? toNote(selectedElement)->string() : -1;
+
             interaction->movePitch(direction, quickly ? PitchMode::OCTAVE : PitchMode::CHROMATIC);
+
+            if (isTab && selectedElement->isNote()) {
+                int nstring = toNote(selectedElement)->string();
+                if (ostring != nstring) {
```

Comment:
Please rename 
`ostring` to `oldString` 
`nstring` to `newString`

And please add new variable `isNote = isTab && selectedElement->isNote()` and reuse